cmd/single.go

```diff
@@ -37,15 +38,21 @@ Beware! May take days to complete.`,
 
 		// Initiate a channel of repositories.
 		repositories := make(chan crawler.Repository, 100)
+		// Prepare WaitGroup.
+		var wg sync.WaitGroup
 
 		// Process each domain service.
 		for _, domain := range domains {
 			if domain.Id == domainID {
-				go crawler.ProcessDomain(domain, repositories)
+				wg.Add(1)
```

Comment:
why we need to count also the domains in the WaitGroup?

Reply:
Special case: after filling the repositories channel with repos all the domains stop to reply to our requests. All the single files are checked and the WG is empty. End.
This wg.Add() will perform an additional check on the domain level. Does not stop until the end is reached (and retry continuously until a domain ends.)